scripts/configuration.js

```diff
@@ -89,36 +56,38 @@ async function populateBuildBranchDropdown(definitionRepository, codeClient, set
  */
 async function onBuildDropdownChange(definitions, codeClient)
 {
+
     console.debug("Resetting the branch dropdown");
-    branchDropdown.innerHTML = defaultBranchDropdownOptions;
-    branchDropdown.value = "";
-    let buildDefinition = buildDropdown.value
+    branchDropdown.innerHTML = '<option value="all" selected>all</option>';
+    branchDropdown.value = "all";
+
 
+    let buildDefinition = buildDropdown.value
     let currentDefinition = definitions.find(d => Number(d.id) === Number(buildDefinition));
 
-    let definitionRepository = currentDefinition.repository.id;
+    console.debug(`Retrieving the branches for the repository ${currentDefinition.repository.id}`);
+    fillBranchesDropDown(currentDefinition.repository.id, codeClient);
+}
```

Comment:
This triggers the configuration change

```suggestion
let definition = definitions.filter(definition => Number(definition.id) === Number(buildDropdown.value))[0];
    let customSettings = {
        data: JSON.stringify({
            buildDefinition: buildDropdown.value,
            buildBranch: branchDropdown.value,
            definitionName: definition.name,
            buildCount: buildCountInput.value,
            defaultTag: defaultTagDropdown.value,
            showStages: showStagesDropdown.value,
        })
    };
    let eventName = WidgetHelpers.WidgetEvent.ConfigurationChange;
    let eventArgs = WidgetHelpers.WidgetEvent.Args(customSettings);
    widgetConfigurationContext.notify(eventName, eventArgs);
}
```